src/main/java/vg/civcraft/mc/civmodcore/CoreConfigManager.java

```diff
@@ -22,57 +31,186 @@ public CoreConfigManager(ACivMod plugin) {
+		this.plugin.info("Logging replies: " + (this.logReplies ? "enabled" : "disabled"));
+		// Allow child class parsing
 		boolean worked = parseInternal(config);
 		if (worked) {
-			plugin.info("Successfully parsed config file of " + plugin.getName());
-		} else {
-			plugin.info("Failed to parse config file of " + plugin.getName() + ". Errors were encountered");
+			plugin.info(ChatColor.BLUE + "Config parsed.");
+		}
+		else {
+			plugin.warning("Failed to pass config!");
 		}
 		return worked;
 	}
 
-	public boolean isDebugEnabled() {
-		return debug;
+	/**
+	 * An internal parser method intended to be overriden by child classes.
+	 *
+	 * @param config The root config section.
+	 * @return Return true if the
+	 */
+	protected boolean parseInternal(ConfigurationSection config) {
+		return true;
+	}
+
+	/**
+	 * This should reset all config values back to their defaults. Child classes should override this if they parse
+	 * additional values that should be reset.
+	 */
+	public void reset() {
+		this.debug = false;
+		this.logReplies = false;
 	}
 
-	public boolean logReplies() {
-		return logReplies;
+	// ------------------------------------------------------------ //
+	// Getters
+	// ------------------------------------------------------------ //
+
+	public final boolean isDebugEnabled() {
+		return this.debug;
+	}
+
+	public final boolean logReplies() {
+		return this.logReplies;
 	}
 
-	protected static <T> List<T> parseList(ConfigurationSection config, String key, Function<String, T> function) {
+	// ------------------------------------------------------------ //
+	// Predefined parsing utilities
+	// ------------------------------------------------------------ //
+
+	/**
+	 * Attempts to parse an integer from a config section, allowing for multiple keys to be attempted.
+	 *
+	 * @param config The config section to retrieve the integer from.
+	 * @param fallback The value that should be falled back upon if a value cannot be found.
+	 * @param keys The keys to attempt.
+	 * @return Returns a retrieved value, or the fallback, which is allowed to be null.
+	 */
+	protected final Integer parseInteger(ConfigurationSection config, Integer fallback, String... keys) {
+		if (config == null || Iteration.isNullOrEmpty(keys)) {
+			return fallback;
+		}
+		for (String key : keys) {
+			if (Strings.isNullOrEmpty(key)) {
+				continue;
+			}
+			if (!config.isInt(key)) {
+				continue;
+			}
+			return config.getInt(key);
+		}
+		return fallback;
+	}
+
+	/**
+	 * Attempts to create a managed database connection with a config section. This is more robust than the default
+	 * method of simply casting a get from the config.
+	 *
+	 * @param config The config section itself. {@code ManagedDatasource db = parseDatabase(config.get("database"));}
+	 * @return Returns an instance of ManagedDatasource, or null if an error occurred while parsing.
+	 */
+	protected final ManagedDatasource parseDatabase(ConfigurationSection config) {
```

Comment:
This defeats the all we have been going for with ManagedDatasource as ConfigurationSerializable and config standardization. I'm heavily opposed.

Reply:
Good, ManagedDatasource's implementation of ConfigurationSerializable is absolutely awful and cannot gracefully handle missing data. It either works and connects, or it errors. And this stuff is parsed (as in a new database connection attempted) *every single time* the config is parsed. This is terrible, it should never have been added.

Reply:
it's so bad you actually have to provide the plugin name within the config of the plugin

Reply:
Removed